contracts/TokenVesting.sol

```diff
@@ -71,286 +98,336 @@ contract TokenVesting is Ownable, ReentrancyGuard{
      */
     constructor(address token_) {
         require(token_ != address(0x0));
-        _token = IERC20(token_);
+        _token = ERC20(token_);
     }
 
-    receive() external payable {}
+    /************************************************
+     *  GETTERS
+     ***********************************************/
 
-    fallback() external payable {}
+    function getCurrentTime() internal view virtual returns (uint256) {
+        return block.timestamp;
+    }
 
     /**
-    * @dev Returns the number of vesting schedules associated to a beneficiary.
-    * @return the number of vesting schedules
-    */
-    function getVestingSchedulesCountByBeneficiary(address _beneficiary)
-    external
-    view
-    returns(uint256){
-        return holdersVestingCount[_beneficiary];
+     * @dev Returns the address of the ERC20 token managed by the vesting contract.
+     */
+    function getToken() external view returns (address) {
+        return address(_token);
     }
 
     /**
-    * @dev Returns the vesting schedule id at the given index.
-    * @return the vesting id
-    */
-    function getVestingIdAtIndex(uint256 index)
-    external
-    view
-    returns(bytes32){
-        require(index < getVestingSchedulesCount(), "TokenVesting: index out of bounds");
+     * @dev Returns the vesting schedule id at the given index.
+     * @return the vesting id
+     */
+    function getVestingIdAtIndex(
+        uint256 index
+    ) external view returns (bytes32) {
+        require(index < vestingSchedulesIds.length, "index out of bounds");
```

Comment:
Do we need this? Solidity will throw for out of bound memory access

Reply:
Verbosity.

Reply:
not saving gas this time?

Reply:
Why saving gas on an external view function?

Reply:
Deployment costs more gas?

Reply:
Use custom error then.